samza-sql/src/main/java/org/apache/samza/sql/translator/QueryTranslator.java

```diff
@@ -165,19 +167,19 @@ private void updateMetrics(long beginProcessing, long endProcessing, SamzaSqlRel
       long outputTime = System.nanoTime();
       queryOutputEvents.inc();
       /* TODO: remove scanTime validation once code to assign it is stable */
-      Validate.isTrue(metadata.hasScanTime());
-      long scanTime = metadata.getscanTime();
+      Validate.isTrue(metadata.hasScanSystemNanoTime());
+      long scanTime = metadata.getScanSystemNanoTime();
       queryLatency.update(TranslatorUtils.getMicroDurationFromNanoTimes(scanTime, outputTime));
-      /** TODO: change if hasArrivalTime to validation once arrivalTime is assigned,
+      /** TODO: change if hasArrivalSystemNanoTime to validation once arrivalTime is assigned,
        and later remove the check once code is stable */
-      if (metadata.hasArrivalTime()) {
-        long arrivalTime = metadata.getarrivalTime();
+      if (metadata.hasArrivalSystemNanoTime()) {
+        long arrivalTime = metadata.getArrivalSystemNanoTime();
         queueingLatency.update(TranslatorUtils.getMicroDurationFromNanoTimes(arrivalTime, scanTime));
       }
       /* since availability of eventTime depends on source, we need the following check */
       if (metadata.hasEventTime()) {
         long eventTime = metadata.getEventTime();
-        totalLatency.update(TranslatorUtils.getMicroDurationFromNanoTimes(eventTime, outputTime));
+        totalLatency.update(Duration.between(Instant.ofEpochMilli(eventTime), Instant.now()).toMillis());
```

Comment:
It looks like this was in micros before, and now you are changing to millis. Looks like this is intentional, but wanted to double check.

Reply:
yes, since we decided to revert back eventTime to EpochMilli